front/src/Turtle/index.test.js

```diff
@@ -31,32 +30,28 @@ describe("Turtle", () => {
         role: "Goalkeeper",
         team: "Cyan"
       };
-      onChangeSpy = sinon.spy();
-      wrapper = shallow(
-        <Turtle
-          turtle={turtle}
-          editable={true}
-          onChange={(...args) => onChangeSpy(args)}
-        />
-      );
+      wrapper = shallow(<Turtle turtle={turtle} editable />);
+      global.fetch = jest.fn().mockImplementation((url, params) => {
+        expect(url).toBe("/api/v1/turtles/");
+        expect(params).toMatchSnapshot();
+        return Promise.resolve({ ok: true });
+      });
+    });
+
+    afterEach(() => {
+      global.fetch = realFetch;
     });
 
     it("should change the role", () => {
       wrapper.find("#turtle2__role").simulate("change", "INACTIVE");
-      expect(onChangeSpy.calledOnce);
-      expect(onChangeSpy.calledWithExactly("role", "INACTIVE"));
     });
```

Comment:
nothing here?

Reply:
The expect calls are actually in the mock implementation in beforeEach